pkg/reconciler/route/resources/labels/labels.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package labels
+
+import (
+	"github.com/knative/serving/pkg/reconciler/route/config"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IsObjectLocalVisibility returns whether an ObjectMeta is of cluster-local visibility
+func IsObjectLocalVisibility(meta v1.ObjectMeta) bool {
+	return meta.Labels != nil && meta.Labels[config.VisibilityLabelKey] != ""
+}
+
+// SetVisibility sets the visibility on an ObjectMeta
+func SetVisibility(meta *v1.ObjectMeta, isClusterLocal bool) {
```

Comment:
@andrew-su Should `SetVisibility` and `SetLabel` belong under `route`? If I'm understanding this correctly, they are not modifying route specific attributes

Reply:
In the original implementation this took in a route object. I later switched it to just `ObjectMeta`. As this is currently the only usages of this is in the route reconciler, I didn't move it. Is there somewhere you think is a better fit?

Reply:
@andrew-su ah I see. How about moving `labels` under [`pkg/reconciler/service`](https://github.com/knative/serving/tree/master/pkg/reconciler/service)?

Reply:
I don't think I agree with moving it to `pkg/reconciler/service` as that's specific to the knative service; the usage is currently used by route, it doesn't make sense to me to reach into another reconciler's package to get this functionality. I can see it living in maybe under `pkg`. Perhaps even waiting until the functionality is required elsewhere to refactor it out to a more common location.

Reply:
@andrew-su I see. Initially I thought it was a bit odd since it's operating on the service object, but was living under `pkg/reconciler/route`. I can't think of a good destination for it now. Like you said, we can move it to another location in the future